.github/workflows/ci-repo.yml

```diff
@@ -11,71 +11,98 @@ on:
       - "main"
       - "release-*"
 
-  workflow_dispatch:
-
-  workflow_call:
-
 concurrency:
   group: ci-repo-${{ github.event_name }}-${{ github.ref }}
   cancel-in-progress: true
 
 jobs:
-  package-tests:
-    runs-on: ubuntu-latest
+  build:
+    name: Build and Package (${{ matrix.arch }})
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        include:
+          - os: ubuntu-latest
+            arch: linux
+          - os: macos-latest
+            arch: macos
+          - os: windows-latest
+            arch: windows
+      max-parallel: 3
 
     steps:
-      - name: Check-out repository
+      - name: Checkout code
         uses: actions/checkout@v4
+        with:
+          ref: ${{ github.ref }}
 
-      - name: Use Node.js from .nvmrc
+      - name: Set up Node.js
         uses: actions/setup-node@v4
         with:
           node-version-file: "vscode/.nvmrc"
 
-      - name: Cache extension node_modules
```

Comment:
I would keep the cache. The pattern of using a separate action is taken from [Continue workflows](https://github.com/continuedev/continue/blob/70d9bc5d52f328d94867a7eb08a41ace3ad06c17/.github/workflows/ts-check.yaml#L24). If you prefer we could also use caching via `setup-node` action.

Reply:
added it back. Thanks